### PR TITLE
🐛 FIX: Ensure removal of "dangling" entity rows

### PR DESCRIPTION
**Pull request by chrisjsewell**
In #5396 (now fixed) it was found that, for sqlalchemy, the deletion of a computer was not removing the related `AuthInfo`, but instead converting the `dbcomputer_id` foreign key to `null`.
Subsequently, in #5379, @mbercx noted that a migration failed, due to a `db_dbgroup_dbnodes` with a `null` `dbnode_id` foreign key.

For such child entities (`db_dbauthinfo`, `db_dbgroup_dbnodes` and also `db_dbcomment`) the deletion of a parent entity, should also remove their rows, rather than leaving them dangling.
The latest PostgreSQL schema, enforces non-null values for these foreign keys.

This PR
- add two additional tests, to ensure both `db_dbgroup_dbnodes` and `db_dbcomment` entries are indeed removed, when removing a node
- Removes any dangling entities, when  migrating from a legacy sqlalchemy database.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
and yes, I have run `tox tests/storage/psql_dos/migrations`, which are no longer run against PRs 😉

File: tests/orm/test_comments.py [hunk 1]
```diff
@@ -194,3 +195,11 @@ def test_objects_get(self):
         comment = node.add_comment('Check out the comment on _this_ one')
         gotten_comment = Comment.objects.get(id=comment.id)
         assert isinstance(gotten_comment, Comment)
+
+    def test_delete_node_with_comments(self):
+        """Test deleting a node with comments."""
+        node = orm.Data().store()
+        node.add_comment('Check out the comment on _this_ one')
+        assert len(Comment.objects.all()) == 2
```
Comment (sphuber) [inline tests/orm/test_comments.py:203]: This seems fragile, relying on the fact that one comment already exists. Would be better to clean the database first or change the check to directly check for the existence of the explicit comment created.

File: tests/orm/test_comments.py [hunk 2]
```diff
@@ -194,3 +195,11 @@ def test_objects_get(self):
...
+        delete_nodes([node.pk], dry_run=False)
+        assert len(Comment.objects.all()) == 1
```
Comment (chrisjsewell) [inline tests/orm/test_comments.py:205]:
```suggestion
    def test_delete_node_with_comments(self):
        """Test deleting a node with comments."""
        assert len(Comment.objects.all()) == 1
        delete_nodes([self.node.pk], dry_run=False)
        assert len(Comment.objects.all()) == 0
```

File: tests/orm/test_comments.py (hunk 1 above)
Comment (chrisjsewell) [reply tests/orm/test_comments.py]: The database is cleaned before each test, then a node and comment is added (in ` def init_profile`, this fixture is run before every test method in the `TestComment` class).
But anyhow, I've just removed the creation of an additional node/comment, and check for the deletion of the one created in `init_profile`

**Review by sphuber — APPROVED** [review]
Thanks @chrisjsewell

**Pull request merged**